src/curve/jacobian.rs

```diff
@@ -0,0 +1,1859 @@
+                c3: Fp(0x430aad2ce14759a4),
+                c4: Fp(0x534ece54de4b2c8),
+                c5: Fp(0xb39050f01f7b1f33),
+            },
+            y: Fp6 {
+                c0: Fp(0xd57f0d0d47482534),
+                c1: Fp(0x26821d894fa8ea0f),
+                c2: Fp(0xc77f564783ef13a1),
+                c3: Fp(0x949c360784284ec2),
+                c4: Fp(0xb7040bd639ef3cc4),
+                c5: Fp(0x8aa635f2719d255f),
+            },
+            z: Fp6::one(),
+        }
+    }
+
+    /// Outputs a compress byte representation of this `JacobianPoint` element
+    pub fn to_compressed(&self) -> CompressedPoint {
+        AffinePoint::from(self).to_compressed()
+    }
+
+    /// Outputs an uncompressed byte representation of this `JacobianPoint` element
+    /// It is twice larger than when calling `JacobianPoint::to_uncompress()`
+    pub fn to_uncompressed(&self) -> UncompressedPoint {
+        AffinePoint::from(self).to_uncompressed()
+    }
+
+    /// Attempts to deserialize an uncompressed element.
+    pub fn from_uncompressed(uncompressed_point: &UncompressedPoint) -> CtOption<Self> {
+        AffinePoint::from_uncompressed(uncompressed_point)
+            .and_then(|p| CtOption::new(JacobianPoint::from(p), 1.into()))
+    }
+
+    /// Attempts to deserialize an uncompressed element, not checking if the
+    /// element is on the curve and not checking if it is in the correct subgroup.
+    /// **This is dangerous to call unless you trust the bytes you are reading; otherwise,
+    /// API invariants may be broken.** Please consider using `from_uncompressed()` instead.
+    pub fn from_uncompressed_unchecked(uncompressed_point: &UncompressedPoint) -> CtOption<Self> {
+        AffinePoint::from_uncompressed_unchecked(uncompressed_point)
+            .and_then(|p| CtOption::new(JacobianPoint::from(p), 1.into()))
+    }
+
+    /// Attempts to deserialize a compressed element.
+    pub fn from_compressed(compressed_point: &CompressedPoint) -> CtOption<Self> {
+        AffinePoint::from_compressed(compressed_point).map(JacobianPoint::from)
+    }
+
+    #[allow(unused)]
+    /// Constructs a `JacobianPoint` element without checking that it is a valid point.
+    pub const fn from_raw_coordinates(elems: [Fp6; 3]) -> Self {
+        JacobianPoint {
+            x: elems[0],
+            y: elems[1],
+            z: elems[2],
+        }
+    }
+
+    /// Computes `n` iterated doubling of this point.
+    pub fn double_multi(&self, n: u32) -> JacobianPoint {
+        assert!(n >= 1);
+        let mut output = self.double();
+
+        for _ in 1..n {
+            output = output.double();
+        }
+
+        output
+    }
+
+    /// Computes `n` iterated doubling of this point.
+    /// **This is dangerous to call unless you know that the point to be doubled
+    /// is not the identity point, otherwise, API invariants may be broken.**
+    /// Please consider using `double()` instead.
```

Comment:
Minor question: should it be "Please consider using `double_multi()` instead." here?